parts/linux/cloud-init/nodecustomdata.yml

```diff
@@ -418,6 +418,110 @@ write_files:
     iptables -I FORWARD -d 168.63.129.16 -p tcp --dport 80 -j DROP
     #EOF
 
+- path: /etc/sysctl.d/999-sysctl-aks.conf
+  permissions: "0644"
+  owner: root
+  content: |
+    # This is a partial workaround to this upstream Kubernetes issue: */}}
+    # https://github.com/kubernetes/kubernetes/issues/41916#issuecomment-312428731 */}}
+    net.ipv4.tcp_retries2=8
+    net.ipv4.tcp_max_syn_backlog=16384
+    net.core.somaxconn=16384
+    net.core.message_burst=80
+    net.core.message_cost=40
+    net.ipv4.neigh.default.gc_thresh1=4096
+    net.ipv4.neigh.default.gc_thresh2=8192
+    net.ipv4.neigh.default.gc_thresh3=16384 
+{{if ShouldConfigSysctl}}
+    # The following are sysctl configs passed from API
+{{- $s:=.CustomLinuxOSConfig.Sysctls}}
+{{- if $s.NetCoreSomaxconn}}
+    net.core.somaxconn={{$s.NetCoreSomaxconn}}
+{{- end}}
+{{- if $s.NetCoreNetdevMaxBacklog}}
+    net.core.netdev_max_backlog={{$s.NetCoreNetdevMaxBacklog}}
+{{- end}}
+{{- if $s.NetCoreRmemMax}}
+    net.core.rmem_max={{$s.NetCoreRmemMax}}
+{{- end}}
+{{- if $s.NetCoreWmemMax}}
+    net.core.wmem_max={{$s.NetCoreWmemMax}}
+{{- end}}
+{{- if $s.NetCoreOptmemMax}}
+    net.core.optmem_max={{$s.NetCoreOptmemMax}}
+{{- end}}
+{{- if $s.NetIpv4TcpMaxSynBacklog}}
+    net.ipv4.tcp_max_syn_backlog={{$s.NetIpv4TcpMaxSynBacklog}}
+{{- end}}
+{{- if $s.NetIpv4TcpMaxTwBuckets}}
+    net.ipv4.tcp_max_tw_buckets={{$s.NetIpv4TcpMaxTwBuckets}}
+{{- end}}
+{{- if $s.NetIpv4TcpFinTimeout}}
+    net.ipv4.tcp_fin_timeout={{$s.NetIpv4TcpFinTimeout}}
+{{- end}}
+{{- if $s.NetIpv4TcpKeepaliveTime}}
+    net.ipv4.tcp_keepalive_time={{$s.NetIpv4TcpKeepaliveTime}}
+{{- end}}
+{{- if $s.NetIpv4TcpKeepaliveProbes}}
+    net.ipv4.tcp_keepalive_probes={{$s.NetIpv4TcpKeepaliveProbes}}
+{{- end}}
+{{- if $s.NetIpv4TcpkeepaliveIntvl}}
+    net.ipv4.tcp_keepalive_intvl={{$s.NetIpv4TcpkeepaliveIntvl}}
+{{- end}}
+{{- if $s.NetIpv4TcpRmem}}
+    net.ipv4.tcp_rmem={{$s.NetIpv4TcpRmem}}
+{{- end}}
+{{- if $s.NetIpv4TcpWmem}}
+    net.ipv4.tcp_wmem={{$s.NetIpv4TcpWmem}}
+{{- end}}
+{{- if $s.NetIpv4TcpTwReuse}}
+    net.ipv4.tcp_tw_reuse={{BoolPtrToInt $s.NetIpv4TcpTwReuse}}
+{{- end}}
+{{- if $s.NetIpv4IpLocalPortRange}}
+    net.ipv4.ip_local_port_range={{$s.NetIpv4IpLocalPortRange}}
+{{- end}}
+{{- if $s.NetIpv4NeighDefaultGcThresh1}}
+    net.ipv4.neigh.default.gc_thresh1={{$s.NetIpv4NeighDefaultGcThresh1}}
+{{- end}}
+{{- if $s.NetIpv4NeighDefaultGcThresh2}}
+    net.ipv4.neigh.default.gc_thresh2={{$s.NetIpv4NeighDefaultGcThresh2}}
+{{- end}}
+{{- if $s.NetIpv4NeighDefaultGcThresh3}}
+    net.ipv4.neigh.default.gc_thresh3={{$s.NetIpv4NeighDefaultGcThresh3}}
+{{- end}}
```

Comment:
same here - can we merge it with the default setting.

Reply:
I don't think we need to. For the same key, later will override the previous setting. Although there could be duplicates but it makes clear what are values we set and what are customer settings. The code is also a bit cleaner.
I added a comment explaining it.

Reply:
why do we need to expose that to cx? What if there is no overriding logic?

Reply:
updated